lib/logstash/inputs/http_poller.rb

```diff
@@ -164,7 +185,14 @@ def handle_success(queue, name, request, response, execution_time)
     # responses come up as "" which will cause the codec to not yield anything
     if body && body.size > 0
       decode_and_flush(@codec, body) do |decoded|
-        event = @target ? LogStash::Event.new(@target => decoded.to_hash) : decoded
+        event = if @target.nil?
+          decoded
+        else
+          e = LogStash::Event.new
+          e.set(@target, decoded.to_hash)
+          e
```

Comment:
:crying_cat_face: that we can not get rid of this - there isn't a better way unless the user sets the `target => ...` for the codec

anyway since codecs will use out [`EventFactorySupport`](https://github.com/logstash-plugins/logstash-mixin-event_support) we should hook it up here as well and use it as : 
```ruby
    include LogStash::PluginMixins::EventSupport::EventFactoryAdapter
```
```suggestion
          targeted_event_factory.new_event(decoded.to_hash)
```

Reply:
yes. I was searching a way to retarget in the codec, so `event.original` can locate properly instead of now nested in `target` field, but I didn't find an easy way to do it, except setting `target` in codec

Reply:
yy and since not all codecs support `target` this is the best we could do ...
the codecs in general will set the `event.original` so for the `target => nil` case we should be good.